packages/angular-material/src/layouts/array-layout.renderer.ts

```diff
@@ -0,0 +1,145 @@
+  THE SOFTWARE.
+*/
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  JsonFormsAngularService,
+  JsonFormsArrayControl
+} from '@jsonforms/angular';
+import {
+  ArrayControlProps,
+  createDefaultValue,
+  generateDefaultUISchema,
+  isObjectArray,
+  mapDispatchToArrayControlProps,
+  OwnPropsOfRenderer,
+  Paths,
+  RankedTester,
+  rankWith,
+  setReadonly,
+  UISchemaElement,
+  unsetReadonly
+} from '@jsonforms/core';
+
+@Component({
+  selector: 'app-array-layout-renderer',
+  template: `
+    <div fxLayout="column" fxLayoutGap="16px" [fxHide]="hidden">
+      <mat-toolbar>
+        <h2>{{ this.label }}</h2>
+        <span fxFlex></span>
+        <button
+          mat-icon-button
+          [disabled]="!isEnabled()"
+          (click)="add()"
+          attr.aria-label="{{ 'Add to ' + this.label + 'button' }}"
+        >
+          <mat-icon>add</mat-icon>
+        </button>
+      </mat-toolbar>
+      <p *ngIf="noData">No data</p>
+      <div
+        *ngFor="
+          let item of data;
+          let idx = index;
+          trackBy: trackByFn;
+          last as last
+        "
+      >
+        <mat-card>
+          <mat-card-content>
+            <jsonforms-outlet [renderProps]="getProps(idx)"></jsonforms-outlet>
+          </mat-card-content>
+          <mat-card-actions *ngIf="isEnabled()">
+            <button mat-button (click)="remove(idx)" attr.aria-label="Remove">
+              <mat-icon>remove_circle</mat-icon>
+              Remove
+            </button>
+          </mat-card-actions>
+        </mat-card>
+      </div>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ArrayLayoutRenderer extends JsonFormsArrayControl {
+  private _uischema: UISchemaElement;
+  noData: boolean;
+  addItem: (path: string, value: any) => () => void;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+
+  constructor(private jsonFormAngularService: JsonFormsAngularService) {
+    super(jsonFormAngularService);
+  }
+  remove(index: number): void {
+    this.removeItems(this.propsPath, [index])();
+  }
+  add(): void {
+    this.addItem(this.propsPath, createDefaultValue(this.scopedSchema))();
+  }
+  ngOnInit() {
+    super.ngOnInit();
+    const { addItem, removeItems } = mapDispatchToArrayControlProps(
+      this.jsonFormAngularService.updateCore.bind(this.jsonFormAngularService)
+    );
+    this.addItem = addItem;
+    this.removeItems = removeItems;
+  }
+
+  mapAdditionalProps(props: ArrayControlProps) {
+    this.noData =
+      !props.data || (Array.isArray(props.data) && !props.data.length);
+  }
+
+  get defaultUISchema() {
+    if (!this._uischema) {
+      this._uischema = generateDefaultUISchema(this.scopedSchema);
+    }
+    return this._uischema;
+  }
+
+  getProps(index: number): OwnPropsOfRenderer {
+    const uischema =
+      (this.uischema.options?.detailUISchema as UISchemaElement) ??
```

Comment:
we should unify this with the react renderer and use `detail`

Reply:
or even better use the findschema like here: https://github.com/eclipsesource/jsonforms/blob/6cf39efbb82fdf77b173c38ca4b706d526920f9b/packages/material/src/layouts/ExpandPanelRenderer.tsx#L101